classes/watcher/login_watcher.php

```diff
@@ -0,0 +1,63 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_mfa\watcher;
+
+defined('MOODLE_INTERNAL') || die();
+
+use core\hook\after_require_login as after_require_login_hook;
+use core\hook\after_config as after_config_hook;
+
+if (class_exists('\totara_core\hook\base')) {
```

Comment:
Could we flip the bit here and make it return early instead? The benefit in doing so is so the class and any extra code blocks are indented at the same levels as other files, and if conditions change or is removed in the future, it won't cause a big diff that is effectively only indentation changes.

```
if (!class_exists('\totara_core\hook\base')) {
     return;
}
```

The same could similarly apply for hooks.php which would keep $watchers at the edge instead of indented due to the surrounding conditional block.

Reply:
Done for both touched files, I'm not sure if short-circuiting has any effect be it autoloaded files or included ones? Let me know if so and I'll revert `db/hooks.php`